### PR TITLE
Add new handler to set the api_id value to the transport headers

### PR DESCRIPTION
**Pull request by denuwanthi**

### CODE REVIEW

File: components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/utils/ValidatorDBUtils.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                "validator.id=subscription_validator.validator_id";
+        String validatorClass = null;
+        try {
+            conn = DbUtils.getDbConnection(DataSourceNames.WSO2TELCO_DEP_DB);
+            ps = conn.prepareStatement(sql);
+            log.debug("getValidatorClassForSubscription for applicationId---> " + applicationId + " apiId--> " + apiId);
+            ps.setInt(1, applicationId);
+            ps.setInt(2, apiId);
+            results = ps.executeQuery();
+            while (results.next()) {
```
Comment (isurulucky) [inline components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/utils/ValidatorDBUtils.java]: What happens if there are no results? Has this been handled?

File: components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/APIInfoHandler.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    public boolean handleRequest(MessageContext messageContext) {
+        AuthenticationContext authContext = APISecurityUtils.getAuthenticationContext(messageContext);
+        int applicationId = Integer.parseInt(authContext.getApplicationId());
+        String api_version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API);
+        api_version = api_version.replace("--", "_").replace(":v", "_");
+        APIIdentifier apiIdentifier = null;
+        try {
+            apiIdentifier = new APIIdentifier(api_version);
+            int apiId = ValidatorDBUtils.getApiId(apiIdentifier);
+            String validatorClassName = getValidatorForSubscription(applicationId, apiId);
```
Comment (isurulucky) [inline components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/APIInfoHandler.java]: IMHO we need to check if we found a proper value for validatorClassName.

**Comment by denuwanthi** [comment]
The check on ResultSet is fixed on commit [4ea806079bc09cf53e2e31f4987a23ab1e067d6d](https://github.com/WSO2Telco/component-dep/pull/164/commits/4ea806079bc09cf53e2e31f4987a23ab1e067d6d)

File: components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/APIInfoHandler.java (hunk 2 above)
Comment (isurulucky) [inline components/validator-handler/src/main/java/com/wso2telco/dep/validator/handler/APIInfoHandler.java]: Great, thanks.

**Pull request merged**